Database/QuestieDB.lua

```diff
@@ -1140,6 +1140,52 @@ function _QuestieDB:HideClassAndRaceQuests()
     Questie:Debug(Questie.DEBUG_DEVELOP, "Other class and race quests hidden");
 end
 
+-- This function is intended for usage with Gossip and Greeting frames, where there's a list of quests but no QuestIDs are
+-- obtainable until entering the specific quest dialog.
+-- This is a bruteforce method for obtaining a QuestID with no input other than a quest name, and the ID of the questgiver.
+-- It compares the name of the quest entry with the names of every quest that questgiver can either start or end.
+-- The GUID that should be passed is UnitGUID("npc").
+-- questStarter should be True if this is an available quest, False if this is an "active" quest (quest ender).
+---@param name string?
+---@param questgiverGUID string?
+---@param questStarter boolean
+---@return number
+function QuestieDB.GetQuestIDFromName(name, questgiverGUID, questStarter)
+    local questID = 0
+    if questgiverGUID then
+        local questgiverID = tonumber(questgiverGUID:match("-(%d+)-%x+$"), 10)
+        local unit_type = strsplit("-", questgiverGUID)
+        local questgiver
+        if unit_type == "Creature" then -- if questgiver is an NPC
+            questgiver = QuestieDB:GetNPC(questgiverID)
```

Comment:
We just need `questStarts` and `questEnds`, don't we? Then it might be more performant to use the `QuestieDB.QueryX` functions as `QuestieDB:GetX` 🤔